File: prototype/rails/src/test/domain/auth/magic_link_token_test.rb
require "test_helper"

module Domain
  module Auth
    class MagicLinkTokenTest < ActiveSupport::TestCase
      def test_digest_is_the_sha256_of_the_token
        assert_equal Digest::SHA256.hexdigest("abc"), MagicLinkToken.digest("abc")
      end

      def test_the_same_token_digests_to_the_same_value
        assert_equal MagicLinkToken.digest("abc"), MagicLinkToken.digest("abc")
      end

      def test_different_tokens_digest_to_different_values
        refute_equal MagicLinkToken.digest("abc"), MagicLinkToken.digest("abd")
      end
    end
  end
end

File: prototype/php/src/app/Domain/Auth/MagicLinkTokenTest.php
<?php

declare(strict_types=1);

namespace App\Domain\Auth;

it('hashes a token with sha256', function (): void {
    expect(MagicLinkToken::hash('abc'))->toBe(hash('sha256', 'abc'));
});

it('hashes the same token to the same digest', function (): void {
    expect(MagicLinkToken::hash('abc'))->toBe(MagicLinkToken::hash('abc'));
});

it('hashes different tokens to different digests', function (): void {
    expect(MagicLinkToken::hash('abc'))->not->toBe(MagicLinkToken::hash('abd'));
});
